bpf/bpf_lxc.c

```diff
@@ -554,6 +554,24 @@ static inline int handle_ipv4_from_lxc(struct __sk_buff *skb, __u32 *dstID)
 			ep_tail_call(skb, CILIUM_CALL_IPV4_NODEPORT_REVNAT);
 			return DROP_MISSED_TAIL_CALL;
 		}
+# ifdef ENABLE_DSR
+		if (ct_state.dsr) {
+			struct ipv4_ct_tuple nat_tup = tuple;
+			struct ipv4_nat_entry *entry;
+
+			nat_tup.flags = NAT_DIR_EGRESS;
+			nat_tup.sport = tuple.dport;
+			nat_tup.dport = tuple.sport;
+			entry = snat_v4_lookup(&nat_tup);
+			if (entry) {
+				ret = snat_v4_rewrite_egress(skb, &nat_tup,
+								  entry, l4_off);
+				if (ret != 0) {
+					return ret;
+				}
```

Comment:
Should likely be an inline helper for nodeport.. same inside ipv4_policy.

Reply:
I will do it as a follow-up.